library/std/src/sys/unix/fd.rs

```diff
@@ -26,6 +27,21 @@ const READ_LIMIT: usize = c_int::MAX as usize - 1;
 #[cfg(not(target_os = "macos"))]
 const READ_LIMIT: usize = libc::ssize_t::MAX as usize;
 
+fn max_iov() -> usize {
+    static LIM: AtomicUsize = AtomicUsize::new(0);
```

Comment:
Are we doing this `LIM` thing just for lazy loading. IIRC std have some sort of `Lazy` that we could use for lazy loading right which may be simpler? Can we use that instead?

Reply:
Please have a look at the resolved discussion at https://github.com/rust-lang/rust/pull/75005#discussion_r463961946 I started out with `std::lazy::SyncOnceCell` but since we do not need only-once semantics here (the `sysconf` value is guaranteed to not change during process lifetime and hence calling at most once per thread is sufficient), using the plain atomic is more efficient.

Reply:
To unpack „simpler“, using OnceCell would allow us to write simpler (shorter) source code, but in terms of what the resulting assembly looks like, atomic version would be much simpler, as it would not include the whole waiting machinery.